public/app/features/alerting/unified/hooks/useProduceNewAlertmanagerConfig.ts

```diff
@@ -0,0 +1,64 @@
+import { Action } from '@reduxjs/toolkit';
+import reduceReducers from 'reduce-reducers';
+
+import { AlertManagerCortexConfig } from 'app/plugins/datasource/alertmanager/types';
+
+import { alertmanagerApi } from '../api/alertmanagerApi';
+import { useAlertmanager } from '../state/AlertmanagerContext';
+
+import { mergeRequestStates } from './mergeRequestStates';
+
+const ERR_NO_ACTIVE_AM = new Error('no active Alertmanager');
+
+const { useLazyGetAlertmanagerConfigurationQuery, useUpdateAlertmanagerConfigurationMutation } = alertmanagerApi;
+
+export const initialAlertmanagerConfiguration: AlertManagerCortexConfig = {
+  alertmanager_config: {
+    receivers: [],
+    route: {},
+    time_intervals: [],
+    mute_time_intervals: [],
+    inhibit_rules: [],
+    templates: [],
+  },
+  template_files: {},
+};
+
+const configurationReducer = reduceReducers(initialAlertmanagerConfiguration);
```

Comment:
This is where we will be adding the individual reducers for receivers, time intervals, notification templates, etc.